scripts/create-camayoc-ocp-config.py

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/env python3
+
+"""Creates a OpenShift configuration for Camayoc.
+
+Given an OpenShift instance with credentials and URL to login,
+this program will create a Openshift configuration for Camayoc.
+
+Example:
+./create-camayoc-ocp-config.py \
+   --name shocp4upi415ovn \
+   --api-url api.shrocp4upi415ovn.lab \
+   --insecure \
+   --auth-token sha256~... \
+   --output config.local.yaml
+"""
+
+import argparse
+import subprocess
+import sys
+
+import yaml
+
+
+def has_oc_command():
+    cmd = ["command", "-v", "oc"]
+    result = subprocess.run(cmd, capture_output=True, check=False, text=True)
+    return True if result.returncode == 0 else False
+
+
+def login(args):
+    url = f"https://{args.api_url}:{args.port}"
+    insecure_tls = "--insecure-skip-tls-verify=true"
+    if args.auth_token:
+        cmd = ["oc", "login", url, insecure_tls, "--token", args.auth_token]
+    else:
+        cmd = ["oc", "login", url, insecure_tls, "-u", args.username, "-p", args.password]
+    result = subprocess.run(cmd, capture_output=True, check=False, text=True)
+    return True if result.returncode == 0 else False
+
+
+def get_auth_token():
+    cmd = ["oc", "whoami", "-t"]
+    result = subprocess.run(cmd, capture_output=True, check=True, text=True)
+    return result.stdout
+
+
+def get_cluster_id():
+    jsonpath = "{$.items[].spec.clusterID}"
+    cmd = ["oc", "get", "clusterversion", "-o", f"jsonpath={jsonpath}"]
+    result = subprocess.run(cmd, capture_output=True, check=True, text=True)
+    return result.stdout
+
+
+def get_cluster_version():
+    jsonpath = "{$.items[*].status.conditions[*].message}"
+    cmd = ["oc", "get", "clusterversion", "-o", f"jsonpath={jsonpath}"]
+    result = subprocess.run(cmd, capture_output=True, check=True, text=True)
+    messages = result.stdout
+    CLUSTER_VERSION_MSG = "Cluster version is "
+    if (start := messages.find(CLUSTER_VERSION_MSG)) == -1:
+        return None
+    # FIXME: fragile, it assumes nothing will follow after the version string
```

Comment:
Thanks, I will do it to improve the code!

Reply:
Another option is `oc version -o json  | jq -r '.openshiftVersion'`

Reply:
Yeah ,looks like `status == True` is the catch (ReleaseAccepted). I'll go check both versions and test if they match or not, just in case.

```
{'lastTransitionTime': '2024-04-05T18:29:44Z', 'message': 'The update channel has not been configured.', 'reason': 'NoChannel', 'status': 'False', 'type': 'RetrievedUpdates'} 
{'lastTransitionTime': '2024-04-05T18:29:44Z', 'message': 'Capabilities match configured spec', 'reason': 'AsExpected', 'status': 'False', 'type': 'ImplicitlyEnabledCapabiliti
es'}                                                                                                                                                                           
{'lastTransitionTime': '2024-04-05T18:29:44Z', 'message': 'Payload loaded version="4.14.0-0.nightly-2024-04-04-140720" image="registry...." architecture="amd64"', 'reason': 'PayloadLoaded', 'status': 'True', 'type': 'ReleaseAccepted'} 
{'lastTransitionTime': '2024-04-05T18:37:59Z', 'message': 'Done applying 4.14.0-0.nightly-2024-04-04-140720', 'status': 'True', 'type': 'Available'}                           
{'lastTransitionTime': '2024-04-05T18:37:59Z', 'message': 'Cluster version is 4.14.0-0.nightly-2024-04-04-140720', 'status': 'False', 'type': 'Progressing'}

```

Reply:
Hey, @abellotti's approach is much more easier, I'm going to use it.